### DISCUSSION TITLE
Segment Fault for CNS Advection

### DISCUSSION THREAD

**Discussion by ChrocheMisawa** [idaholab/moose#17815] [Q&A Modules: Navier-Stokes] [answered]
Hello,

      I'm current using the latest branch from @lindsayad , I was trying to produce some similar results for the model that I shown in #17760 . However, the code gives Segmentation fault error whenever I ran it with multiple cores, both for my input and the test input file at "modules/navier_stokes/test/tests/finite_volume/cns/scalar_advection/mass-frac-advection.i"

      Both input file can run under single thread mode. (Although mine does not converge)
 
      I wonder if the MPI feature is fully supported at this stage.
    
[Flow_Past_a_Cylinder_cns.zip](https://github.com/idaholab/moose/files/6470029/Flow_Past_a_Cylinder_cns.zip)

Thank you,
Zeyu Chen

**Comment by aeslaughter** [comment]
@lindsayad

**Comment by lindsayad** [comment, marked as answer]
I will try to take a look at this soon

**Reply by lindsayad** [reply]
@ChrocheMisawa if you pull my most recent branch updates these should run now. There was an error in the number of ghost layers being used. Previously that input was generating one ghost layer when it needed two.
